pfl/internal/ops/pytorch_ops.py

```diff
@@ -167,7 +167,12 @@ def __init__(self):
                     hvd.local_rank(), hvd.local_size(), hvd.rank(), hvd.size())
         self._post_init_check(hvd)
 
-        if torch.cuda.is_available():
+        if torch.cuda.is_available(
```

Comment:
Do you need an `else` clause if `"CUDA_VISIBLE_DEVICES" not in os.environ`?

Reply:
I don't think so, `CUDA_VISIBLE_DEVICES` handles the set_device in that case.